src/main/java/com/avispl/dal/communicator/pexip/PexipManagementNode.java

```diff
@@ -0,0 +1,929 @@
+     * @throws Exception if any error occurs
+     */
+    private List<Map<String, String>> retrieveParticipants(String conferenceId) throws Exception {
+        JsonNode response = doGet(String.format(CONFERENCE_PARTICIPANTS_URI, 5000, conferenceId), JsonNode.class);
+        return extractParticipants(response);
+    }
+
+    /**
+     * Retrieve participants status as {@link List} of {@link Map}, map instance per participant.
+     * Values are extracted using model-mapping.yml, as AggregatedDevice instances (for v1, targeted to SY 5.1),
+     * v2 will have data exported as {@link Map} (targeted to SY v5.2)
+     *
+     * @return {@link List} of {@link Map} containing {@link String} key:value pairs, representing participants statuses
+     * @throws Exception if any error occurs
+     */
+    private List<Map<String, String>> retrieveParticipants() throws Exception {
+        JsonNode response = doGet(String.format(PARTICIPANTS_URI, 5000), JsonNode.class);
+        return extractParticipants(response);
+    }
+
+    /**
+     * Extract participants as {@link List} of {@link Map}, map instance per participant, from {@link JsonNode} API response.
+     * Values are extracted using model-mapping.yml, as AggregatedDevice instances (for v1, targeted to SY 5.1),
+     * v2 will have data exported as {@link Map} (targeted to SY v5.2)
+     *
+     * @return {@link List} of {@link Map} containing {@link String} key:value pairs, representing participants statuses
+     * @throws Exception if any error occurs
+     */
+    private List<Map<String, String>> extractParticipants(JsonNode response) throws Exception {
+        List<Map<String, String>> participants = new ArrayList<>();
+        ArrayNode participantObjects = (ArrayNode) response.get(OBJECTS);
+
+        if (participantObjects != null && !participantObjects.isEmpty()) {
+            participantObjects.forEach(node -> {
+                /* After 5.2 will be changed to Map<String, String> participantReport = new HashMap<>(); */
+                AggregatedDevice participantReport = new AggregatedDevice();
+                aggregatedDeviceProcessor.applyProperties(participantReport, node, "Participant");
+                participants.add(participantReport.getProperties());
+            });
+        }
+        return participants;
+    }
+
+    @Override
+    public List<AggregatedDevice> retrieveMultipleStatistics() throws Exception {
+        List<Map<String, String>> conferences = retrieveConferencesStatus();
+        List<Map<String, String>> participants = retrieveParticipants();
+        List<AggregatedDevice> conferencingNodes = retrieveConferencingNodes();
+
+        knownConferences.clear();
+        knownParticipants.clear();
+
+        if (!displayParticipantsStatistics && !displayConferencesStatistics) {
+            return conferencingNodes;
+        }
+        conferencingNodes.forEach(aggregatedDevice -> {
+            if (displayConferencesStatistics) {

```

Comment:
Maybe I'm mistaking, but what is the point of going through each conferencingNode, if displayConferencesStatistics is false? I would wrap the forEach in this if.

Reply:
Yes, you are right, the condition above is already doing this, so this check is removed now :)